src/twisted/test/test_defer.py

```diff
@@ -71,160 +98,190 @@ class ImmediateFailureMixin:
     Add additional assertion methods.
     """
 
-    def assertImmediateFailure(self, deferred, exception):
+    def assertImmediateFailure(
+        self, deferred: Deferred[Any], exception: Type[Exception]
+    ) -> Exception:
```

Comment:
This could be made more precise using `ExceptionT = TypeVar("ExceptionT", bound=Exception)`. It doesn't really seem to make a difference in practice though, since there is only one case where the returned value is used and in that case `Exception` is passed as the exception type.

Reply:
Done, thanks!